services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/LocalStatisticsStructureProvider.java

```diff
@@ -0,0 +1,52 @@
+package app.coronawarn.server.services.distribution.assembly.component;
+
+import app.coronawarn.server.common.protocols.internal.stats.LocalStatistics;
+import app.coronawarn.server.services.distribution.assembly.structure.Writable;
+import app.coronawarn.server.services.distribution.assembly.structure.WritableOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.statistics.directory.LocalStatisticsDirectory;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+
+/**
+ * Reads configuration parameters from the respective files in the class path and build a {@link
+ * LocalStatisticsStructureProvider} with them.
+ */
+@Component
+public class LocalStatisticsStructureProvider {
+
+  private final CryptoProvider cryptoProvider;
+  private final DistributionServiceConfig distributionServiceConfig;
+  private final Map<Integer, LocalStatistics> localStatisticsMap;
+
+  /**
+   * Creates a new LocalStatisticsStructureProvider.
+   */
+  public LocalStatisticsStructureProvider(
+      CryptoProvider cryptoProvider,
+      DistributionServiceConfig distributionServiceConfig,
+      Map<Integer, LocalStatistics> localStatisticsMap) {
+    this.cryptoProvider = cryptoProvider;
+    this.distributionServiceConfig = distributionServiceConfig;
+    this.localStatisticsMap = localStatisticsMap;
+  }
+
+  /**
+   * Returns a list containing the archives with Local Statistics for mobile clients using signature file.
+   */
+  public List<Writable<WritableOnDisk>> getLocalStatisticsList() {
```

Comment:
Not sure about the structure here. Do we not normally return a Directory<WriteableOnDisk> when working with multiple data? For example  public Directory<WritableOnDisk> getDiagnosisKeys() {. What is the difference here?

Reply:
Yes, we normally do that but the requirement is for all the archives (ex local_stats_1, local_stats_2, local stats_3 etc) to be in the 'v1' directory and not in their own.